src/ComponentFactory.php

```diff
@@ -79,9 +79,11 @@ protected function requirePath(string $path): void
                         : $variable;
                 }, get_defined_vars());
             })();
+            CompileContext::instance()->variables =  array_merge(CompileContext::instance()->variables, $variables);
```

Comment:
Why is this unrelated change made?

Reply:
@taylorotwell sorry, yep this was needed further down the track of my implementation. The mingle lib needs a `component` method defined to point to the react/vue component it should render, and I wanted to add it automatically as I'm extracting the component from the blade file itself  (see https://github.com/deanmcpherson/mingle/blob/main/src/Mingle.php), but variables added to the default array before this is run were getting overridden here.